### PR TITLE
Add notebook for RunInference with Hugging Face 

### PR DESCRIPTION
**Pull request by riteshghorse**
Adds a notebook with 2 examples of using Hugging Face model handlers.
[Notebook](https://github.com/riteshghorse/beam/blob/hf-notebook/examples/notebooks/beam-ml/run_inference_huggingface.ipynb)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by riteshghorse** [comment]
R: @rszper

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+        "## RunInference with pretrained model from Hugging Face Hub\n"
+      ],
+      "metadata": {
+        "id": "KJEsPkXnUS5y"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "To use pretrained models directly from Hugging Face hub, use the `HuggingFaceModelHandlerTensor` or  `HuggingFaceModelHandlerKeyedTensor` depending upon the input type. Use `load_model_args` and `inference_args` to provide optional arguments to load the model and do the inference respectively. Also, please specify the `framework='tf'` for TensorFlow models and `'pt'` for PyTorch models.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:304]:
```suggestion
        "To use pretrained models directly from [Hugging Face Hub](https://huggingface.co/docs/hub/models), use either the `HuggingFaceModelHandlerTensor` model handler or the `HuggingFaceModelHandlerKeyedTensor` model handler. Which model handler you use depends on your input type:\n",
        "\n",
        "- Use `HuggingFaceModelHandlerKeyedTensor` to run inference with models that uses keyed tensors as inputs.\n",
        "- Use `HuggingFaceModelHandlerTensor` to run inference with models that uses tensor inputs, such as `tf.Tensor` or `torch.Tensor`.\n",
        "\n",
        "When you construct your pipeline, you might also need to use the following items:\n",
        "\n",
        "- Use the `load_model_args` argument to provide optional arguments to load the model.\n",
        "- Use the `inference_args` argument to do the inference.\n",
        "- For TensorFlow models, specify the `framework='tf'`.\n",
        "- For PyTorch models, specify the framework `'pt'`.\n",
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+        "\n",
+        "\n",
+        "Let's do a language modeling task to predict the masked word in a sentence."
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:307]:
```suggestion
        "The following language modeling task predicts the masked word in a sentence."
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "mDcpG78tWcBN"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Create a model handler\n",
+        "\n",
+        "We will do a masked language modeling task. These models take keyed tensors as input."
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:318]:
```suggestion
        "This example shows a masked language modeling task. These models take keyed tensors as inputs."
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "Zx1ep1UXWYrq"
+      },
+      "execution_count": 17,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Define input examples"
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:344]:
```suggestion
        "### Define the input examples\n",
        "\n",
        "Use this code to define the input examples."
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "vWI_A6VrcH-m"
+      },
+      "execution_count": 18,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Preprocess input\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:368]:
```suggestion
        "### Preprocess the input\n",
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+        "\n",
+        "Edit the given input to replace the last word with a `<mask>` and tokenize it for doing inference."
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:370]:
```suggestion
        "Edit the given input to replace the last word with a `<mask>`. Then, tokenize the input for doing inference."
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 7]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "-62nvMSbeNBy"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def add_mask_to_last_word(text: str) -> Tuple[str, str]:\n",
+        "  \"\"\"Replace the last word of sentence with <mask> and return\n",
+        "  the original sentence and masked sentence\"\"\"\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:381]:
```suggestion
        "  the original sentence and the masked sentence.\"\"\"\n",
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 8]
```diff
@@ -0,0 +1,510 @@
...
+        "    text_and_mask: Tuple[str, str],\n",
+        "    tokenizer) -> Tuple[str, Dict[str, tf.Tensor]]:\n",
+        "  \"\"\"Convert string examples to tensors.\"\"\"\n",
+        "  text, masked_text = text_and_mask\n",
+        "  tokenized_sentence = tokenizer.encode_plus(\n",
+        "      masked_text, return_tensors=\"tf\")\n",
+        "\n",
+        "  # Workaround to manually remove batch dim until we have the feature to\n",
+        "  # add optional batching flag.\n",
+        "  # TODO(https://github.com/apache/beam/issues/21863): Remove once optional\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:398]:
```suggestion
        "  # TODO(https://github.com/apache/beam/issues/21863): Remove when optional\n",
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 9]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "d6TXVfWhWRzz"
+      },
+      "execution_count": 19,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Postprocess results\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:414]:
```suggestion
        "### Postprocess the results\n",
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 10]
```diff
@@ -0,0 +1,510 @@
...
+        "\n",
+        "Extract the result from `PredictionResult` object and format output to print the actual sentence and predicted word for last word in the sentence."
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:416]:
```suggestion
        "Extract the result from the `PredictionResult` object. Then, format the output to print the actual sentence and the word predicted for the last word in the sentence."
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 11]
```diff
@@ -0,0 +1,510 @@
...
+        "id": "KnLtuYyTfC-g"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "class PostProcessor(beam.DoFn):\n",
+        "  \"\"\"Processes the PredictionResult to get the predicted word.\n",
+        "\n",
+        "  The logits are the output of the BERT Model. We can get the word with the highest\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:428]:
```suggestion
        "  The logits are the output of the BERT Model. To get the word with the highest\n",
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 12]
```diff
@@ -0,0 +1,510 @@
...
+        "  probability of being the masked word by taking the argmax.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:429]:
```suggestion
        "  probability of being the masked word, take the argmax.\n",
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 13]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "DnWlNV1kelnq"
+      },
+      "execution_count": 28,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Run Pipeline"
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:454]:
```suggestion
        "### Run the pipeline\n",
        "\n",
        "Use the following code to run the pipeline."
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 14]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "aF_BDPXk3UG4"
+      },
+      "execution_count": 13,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Define input examples"
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:197]:
```suggestion
        "### Define the input examples\n",
        "\n",
        "Use this code to define the input examples."
```

File: examples/notebooks/beam-ml/run_inference_huggingface.ipynb [hunk 15]
```diff
@@ -0,0 +1,510 @@
...
+      "metadata": {
+        "id": "POAuIFS_UDgE"
+      },
+      "execution_count": 14,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Postprocess results\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_huggingface.ipynb:218]:
```suggestion
        "### Postprocess the results\n",
```

**Pull request merged**